src/execution_plan/ops/op_update.c

```diff
@@ -111,9 +114,12 @@ OpResult OpUpdateReset(OpBase *ctx) {
 
 /* Executes delayed updates. */
 void _UpdateEntities(OpUpdate *op) {
+    EntityUpdateCtx *entity_to_update;
     for(int i = 0; i < op->entities_to_update_count; i++) {
-        EntityProperty *dest_entity_prop = op->entities_to_update[i].dest_entity_prop;
-        SIValue new_value = op->entities_to_update[i].new_value;
+        entity_to_update = &op->entities_to_update[i];
+        EntityProperty *dest_entity_prop = entity_to_update->dest_entity_prop;
+        SIValue new_value = entity_to_update->new_value;
+        if (entity_to_update->t == N_ENTITY) GraphContext_UpdateNodeIndices(op->gc, entity_to_update->id, dest_entity_prop, &new_value);
```

Comment:
As long as we do not support multiple labels for nodes, accept an optional store / label argument in GraphContext_UpdateNodeIndices